api/onix.py

```diff
@@ -222,7 +222,10 @@ def parse(cls, file, data_source_name, default_medium=None):
             publishing_date = parser.text_of_optional_subtag(record, 'publishingdetail/publishingdate/b306')
             issued = None
             if publishing_date:
-                issued = datetime.datetime.strptime(publishing_date, "%Y%m%d")
+                try:
+                    issued = datetime.datetime.strptime(publishing_date, "%Y%m%d")
+                except ValueError:
+                    issued = datetime.datetime.strptime(publishing_date, "%Y")
```

Comment:
Since ONIX has a variety of supported date formats and we're already being somewhat imprecise, perhaps we could use a more generalized approach that will handle more possibilities:
```
from dateutil.parser import isoparse
...
publishing_date = parser.text_of_optional_subtag(record, 'publishingdetail/publishingdate/b306')
issued = isoparse(publishing_date) if publishing_date else None